zkSNACKs/WalletWasabi
# `JsonRpcServer`: Handle `stop` command better + refactoring

<s>Depends on #10596</s>
<s>Depends on #10598</s>
Related to #10546
Related comment https://github.com/zkSNACKs/WalletWasabi/pull/10529#issuecomment-1518453849

This PR fixes behavior when JSON RPC `stop` command is received. We no longer wait 21 seconds but finish immediately.

## Testing

1. Start WW (either Daemon or Fluent)
2. Run `curl -s --data-binary '{"jsonrpc":"2.0", "method":"stop"}' http://127.0.0.1:37128/` to stop the wallet
3. The wallet should stop immediately.

I tested on Windows, mac, and linux. It worked for me.
